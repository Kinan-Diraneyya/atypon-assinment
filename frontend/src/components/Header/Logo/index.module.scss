.logo {
    color: white;
    text-transform: uppercase;
    font-weight: 200;
    width: min-content;
    text-align: center;
    font-size: 1.2rem;
}