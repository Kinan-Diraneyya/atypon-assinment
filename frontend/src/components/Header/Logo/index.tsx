import style from './index.module.scss';

/**
 * Application logo.
 * 
 * Renders text that represents the application's logo
 * 
 * @returns {JSX.Element} The Logo component.
 */
export default function Logo() {
    return <a href='/' className={style.logo}>Spoonacular Recipe Client</a>;
}