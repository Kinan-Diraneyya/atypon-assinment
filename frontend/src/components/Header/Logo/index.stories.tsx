import React from 'react';
import Logo from './index';
import { Meta, StoryObj } from '@storybook/react';

const meta: Meta<typeof Logo> = {
    component: Logo,
  };
   
  export default meta;
  type Story = StoryObj<typeof Logo>;
   
  export const Main: Story = {
    args: {}
  };