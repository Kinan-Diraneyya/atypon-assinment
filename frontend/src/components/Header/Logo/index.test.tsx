import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Logo from './index';

describe('Logo Component', () => {
  test('renders without crashing', () => {
    render(<Logo />);
    const logoElement = screen.getByText('Spoonacular Recipe Client');
    expect(logoElement).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const { asFragment } = render(<Logo />);
    expect(asFragment()).toMatchSnapshot();
  });
});