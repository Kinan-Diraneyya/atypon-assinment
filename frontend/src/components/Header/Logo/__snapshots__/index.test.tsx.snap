// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="logo"
  >
    Spoonacular Recipe Client
  </div>
</DocumentFragment>
`;
